src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -490,27 +515,42 @@ public SnapshotSummaryModel makeSummaryModelFromSummary(SnapshotSummary snapshot
         return summaryModel;
     }
 
-    private SnapshotModel populateSnapshotModelFromSnapshot(Snapshot snapshot) {
-        return new SnapshotModel()
-                .id(snapshot.getId().toString())
-                .name(snapshot.getName())
-                .description(snapshot.getDescription())
-                .createdDate(snapshot.getCreatedDate().toString())
-                .profileId(snapshot.getProfileId().toString())
-                .source(snapshot.getSnapshotSources()
-                        .stream()
-                        .map(this::makeSourceModelFromSource)
-                        .collect(Collectors.toList()))
-                .tables(snapshot.getTables()
-                        .stream()
-                        .map(this::makeTableModelFromTable)
-                        .collect(Collectors.toList()))
-                .relationships(snapshot.getRelationships()
-                        .stream()
-                        .map(this::makeRelationshipModelFromRelationship)
-                        .collect(Collectors.toList()))
-                .dataProject(snapshot.getProjectResource().getGoogleProjectId());
+    private SnapshotModel populateSnapshotModelFromSnapshot(Snapshot snapshot,
+                                                            List<SnapshotRequestAccessInclude> include) {
+        SnapshotModel snapshotModel = new SnapshotModel()
+            .id(snapshot.getId().toString())
+            .name(snapshot.getName())
+            .description(snapshot.getDescription())
+            .createdDate(snapshot.getCreatedDate().toString());
 
+        if (doInclude(include, SnapshotRequestAccessInclude.SOURCES, true)) {
```

Comment:
Using a `true` / `false` parameter doesn't seem like the best way to represent the default settings. It requires every call to `doInclude()` to pass in a value, and makes it harder to read the code and see what values are enabled by default.

If the only code path to this method where `include` is empty is from the call on line 178 above, it would be better to pass in the default list instead of `Collections.emptyList()`. If there are cases other than that, then this method could check for `include.isEmpty()` and replace it with the default list in that case.

The special form `NONE` could also be handled at the top of this method. Once that's done, the `doInclude()` calls could be replaced with `include.contains()` calls.

Reply:
Yeah, that's a fair point about the default value.  That's an interesting idea about passing that in on empty.  I agree it makes the verification logic simpler.  Ok, I'll update

Reply:
Ah and I did a little digging.  The reason I had done it this way is for backward compatibility (so that existing calls would return the same value) but in fact, Spring properly substitutes the default value in when nothing is passed in with the `include` parameter so we can safely assume that the default will not be an empty list